File: app/src/main/java/com/cylan/jiafeigou/base/injector/component/AppComponent.java
package com.cylan.jiafeigou.base.injector.component;

import android.content.Context;

import com.cylan.jfgapp.interfases.AppCmd;
import com.cylan.jiafeigou.base.injector.lifecycle.ContextLife;
import com.cylan.jiafeigou.base.injector.module.AppModule;
import com.cylan.jiafeigou.base.injector.module.CommonModule;
import com.cylan.jiafeigou.base.module.BaseDeviceInformationFetcher;
import com.cylan.jiafeigou.base.module.BaseInitializationManager;
import com.cylan.jiafeigou.base.module.BasePanoramaApiHelper;
import com.cylan.jiafeigou.base.module.BasePresenterInjector;
import com.cylan.jiafeigou.base.view.IPropertyParser;
import com.cylan.jiafeigou.base.view.JFGPresenter;
import com.cylan.jiafeigou.base.view.JFGSourceManager;
import com.cylan.jiafeigou.cache.db.view.IDBHelper;
import com.cylan.jiafeigou.cache.db.view.IDPTaskDispatcher;
import com.cylan.jiafeigou.misc.pty.IProperty;
import com.cylan.jiafeigou.support.badge.TreeHelper;
import com.danikula.videocache.HttpProxyCacheServer;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Component;
import okhttp3.OkHttpClient;

/**
 * Created by yanzhendong on 2017/4/12.
 */
@Singleton
@Component(modules = {AppModule.class, CommonModule.class})
public interface AppComponent {

    @ContextLife()
    Context getAppContext();

    JFGPresenter getDefaultPresenter();

    JFGSourceManager getSourceManager();

    IDBHelper getDBHelper();

    IDPTaskDispatcher getTaskDispatcher();

    IPropertyParser getPropertyParser();

    BasePresenterInjector getBasePresenterInjector();

    BaseInitializationManager getInitializationManager();

    AppCmd getCmd();

    HttpProxyCacheServer getHttpProxyCacheServer();

    BasePanoramaApiHelper getHttpApiHelper();

    BaseDeviceInformationFetcher getDeviceInformationFetcher();

    OkHttpClient getOkHttpClient();

    IProperty getProductProperty();

    @Named("LogPath")
    String getLogPath();

    TreeHelper getTreeHelper();
}

File: app/src/main/java/com/cylan/jiafeigou/dagger/component/AppComponent.kt
package com.cylan.jiafeigou.dagger.component

import android.content.Context
import com.cylan.jiafeigou.base.module.IHttpApi
import com.cylan.jiafeigou.dagger.annotation.ContextLife
import com.cylan.jiafeigou.dagger.annotation.Named
import com.cylan.jiafeigou.dagger.module.ActivityModule
import com.cylan.jiafeigou.dagger.module.CommonModule
import com.cylan.jiafeigou.dagger.module.FragmentModule
import com.cylan.jiafeigou.n.base.BaseApplication
import com.cylan.jiafeigou.support.badge.TreeHelper
import com.danikula.videocache.HttpProxyCacheServer
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import okhttp3.OkHttpClient
import javax.inject.Singleton


/**
 * Created by yanzhendong on 2017/10/26.
 */
@Singleton
@Component(modules = arrayOf(
        AndroidSupportInjectionModule::class,
        ActivityModule::class,
        FragmentModule::class,
        CommonModule::class
))
interface AppComponent : AndroidInjector<BaseApplication> {

    override fun inject(baseApplication: BaseApplication)
    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: BaseApplication): AppComponent.Builder

        fun build(): AppComponent
    }

    @ContextLife()
    fun getAppContext(): Context

    fun getHttpProxyCacheServer(): HttpProxyCacheServer

    fun getOkHttpClient(): OkHttpClient

    @Named("LogPath")
    fun getLogPath(): String

    fun getTreeHelper(): TreeHelper

    fun getHttpApi(): IHttpApi
}